src/net_processing.cpp

```diff
@@ -3250,7 +3250,17 @@ bool SendMessages(CNode* pto, CConnman& connman, const std::atomic<bool>& interr
         if (state.vBlocksInFlight.size() > 0) {
             QueuedBlock &queuedBlock = state.vBlocksInFlight.front();
             int nOtherPeersWithValidatedDownloads = nPeersWithValidatedDownloads - (state.nBlocksInFlightValidHeaders > 0);
-            if (nNow > state.nDownloadingSince + consensusParams.nPowTargetSpacing * (BLOCK_DOWNLOAD_TIMEOUT_BASE + BLOCK_DOWNLOAD_TIMEOUT_PER_PEER * nOtherPeersWithValidatedDownloads)) {
+            int64_t nCalculatedDlWindow = 
+            std::max(consensusParams.nPowTargetSpacing, MIN_BLOCK_DOWNLOAD_MULTIPLIER) *
+             (BLOCK_DOWNLOAD_TIMEOUT_BASE + 
+             BLOCK_DOWNLOAD_TIMEOUT_PER_PEER * 
+             nOtherPeersWithValidatedDownloads);
+            if (nNow > state.nDownloadingSince + nCalculatedDlWindow) {
+                LogPrint("net", "Timeout downloading block: window=%d; inFlight=%d; validHeaders=%d; otherDlPeers=%d;",
+                 nCalculatedDlWindow, 
+                 state.vBlocksInFlight.size(), 
+                 state.nBlocksInFlightValidHeaders, 
+                 nOtherPeersWithValidatedDownloads);
                 LogPrintf("Timeout downloading block %s from peer=%d, disconnecting\n", queuedBlock.hash.ToString(), pto->id);
                 pto->fDisconnect = true;
```

Comment:
Yes it's intentional when providing -debug=net to gain deeper insight into the values of those 4 variables. The other message is not really helpful regarding debugging, at least to me in that it only provides hash and peer at timeout.

Reply:
I requested this debug message so that when timeouts happen consistently, debug=net actually says something about this rather than just saying hey you timed out, as it's impossible to deduct this state from a node in the field. So the original `LogPrintf` line helps detecting issues when just running a node, the new `LogPrint("net", ...)` helps with actual debugging when things go south in the field.